commands.go

```diff
@@ -369,35 +369,42 @@ func cmdActive(c *cli.Context) {
 	}
 }
 
+// If the user has specified a driver, they should not see the flags for all
+// of the drivers in `docker-machine create`.  This method replaces the 100+
+// create flags with only the ones applicable to the driver specified
+func trimDriverFlags(driver string, cmds []cli.Command) ([]cli.Command, error) {
+	filteredCmds := cmds
+	driverFlags, err := drivers.GetCreateFlagsForDriver(driver)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, cmd := range cmds {
+		if cmd.HasName("create") {
+			filteredCmds[i].Flags = append(sharedCreateFlags, driverFlags...)
```

Comment:
Do you think we should keep the same ordering as above on line 207? (driver flags then shared flags)